test/Microsoft.ML.Tests/Scenarios/Api/Estimators/TrainWithInitialPredictor.cs

```diff
@@ -37,8 +37,8 @@ public void New_TrainWithInitialPredictor()
                 var firstModel = trainer.Fit(trainData);
 
                 // Train the second predictor on the same data.
-                var secondTrainer = new MyAveragedPerceptron(env, new AveragedPerceptronTrainer.Arguments(), "Features", "Label");
-                var finalModel = secondTrainer.Train(trainData, firstModel.Model);
+                var secondTrainer = new AveragedPerceptronTrainer(env, new AveragedPerceptronTrainer.Arguments());
+                var finalModel = secondTrainer.Fit(trainData, initialPredictor: firstModel.Model);
```

Comment:
>Fit [](start = 47, length = 3)

@tfinley@gmail.com if Fit() should not take a validation set, and an initial model, for this example/case here, would we still call Train()?


// Train the second predictor on the same data.
                var secondTrainer = new AveragedPerceptronTrainer(env, new AveragedPerceptronTrainer.Arguments());

                var trainRoles = new RoleMappedData(trainData, label: "Label", feature: "Features");
                var finalModel = secondTrainer.Train(new TrainContext(trainRoles, initialPredictor: firstModel.Model));

Reply:
In Pigsty land, I'm not sure how this can be accommodated (or rather should it?)

In Dynamic land, yes, we still call Train, just not with the TrainContext and RoleMappedData

---
In reply to: [215809082](https://github.com/dotnet/machinelearning/pull/849#discussion_r215809082) [](ancestors = 215809082)

Reply:
> In Pigsty land, I'm not sure how this can be accommodated (or rather should it?)

Almost certainly, but unless I'm mistaken I don't *think* changes to `IEstimator`, or any of these core interfaces, will be necessary, or indeed even slightly helpful. Indeed, I might be missing something, but I'm a little confused about where we imagine the source of the problem is? We've posited the existence of some hypothetical `ITransformer Train(IDataView trainData, ...)` method, so why do we suppose a `Transformer<TIn, TOut> Train<TIn, TOut>(DataView<TIn> trainData, ...)` is impossible? Perhaps there's some subtlety I've missed.

---
In reply to: [215821268](https://github.com/dotnet/machinelearning/pull/849#discussion_r215821268) [](ancestors = 215821268,215809082)

Reply:
I'm not certain whether that's in scope for this PR. Maybe it is, I just don't know.

---
In reply to: [215840658](https://github.com/dotnet/machinelearning/pull/849#discussion_r215840658) [](ancestors = 215840658,215821268,215809082)